src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -872,57 +873,29 @@ public AppPage verifyContains(String searchString) {
         
     /**
      * Verifies the status message in the page is same as the one specified.
-     * @return The page (for chaining method calls).
      */
-    public AppPage verifyStatus(String expectedStatus){
+    public void verifyStatus(String expectedStatus) {
         
-        try{
-            assertEquals(expectedStatus, this.getStatus());
-        } catch(Exception e){
-            if(!expectedStatus.equals("")){
-                this.waitForElementPresence(By.id("statusMessagesToUser"));
-                if(!statusMessage.isDisplayed()){
-                    this.waitForElementVisibility(statusMessage);
+        // The check is done multiple times with waiting times in between to account for
+        // timing issues due to page load, inconsistencies in Selenium API, etc.
+        for (int i = 0; i < VERIFICATION_RETRY_COUNT; i++) {
+            try {
+                waitForElementVisibility(statusMessage);
+                if (expectedStatus.equals(getStatus())) {
+                    break;
                 }
+            } catch (StaleElementReferenceException e) {
+                // Might occur if the page reloads, which makes the previous WebElement
+                // stored in the variable statusMessage "stale"
             }
-        }
-        assertEquals(expectedStatus, this.getStatus());
-        return this;
-    }
-
-    /**
-     * Verifies the status message in the page is same as the one specified, retrying up till numberOfTries
-     * times. This can be used when status message test results in inconsistency due to timing issues
-     * or inconsistency in Selenium's API such as .click() which may or may not result in detecting a page
-     * load depending on things such as browser type and how an event is triggered.
-     * 
-     * Note that we must wait for the next page's status message to be visible on every try within the while
-     * loop. This is because the previous check might have checked early and have not waited for the page
-     * load to finish. From that, it is possible to create a situation whereby the page has now loaded
-     * on the next try (after 1000ms) and the status message is still loading and still not displayed.
-     * 
-     * @return The app page itself for method chaining flexibility
-     */
-    public AppPage verifyStatusWithRetry(String expectedStatus, int maxRetryCount) {
-        int currentRetryCount = 0;
-
-        while (currentRetryCount < maxRetryCount) {
-            if (!statusMessage.isDisplayed()) {
-                this.waitForElementVisibility(statusMessage);
-            }
-
-            if (expectedStatus.equals(this.getStatus())) {
+            if (i == VERIFICATION_RETRY_COUNT - 1) {
```

Comment:
I think this if could be the first thing inside the loop to maintain the 5 tries. (just following the verifyHtml example).